noir-projects/aztec-nr/aztec/src/context/private_context.nr

```diff
@@ -405,7 +405,16 @@ impl PrivateContext {
             caller_context.storage_contract_address = self.inputs.call_context.storage_contract_address;
         }
         self.private_call_requests.push(
-            PrivateCallRequest { hash: item.hash(), caller_context, start_side_effect_counter, end_side_effect_counter }
+            PrivateCallRequest {
```

Comment:
Could it make sense for the oracle to return directly the items necessary for the PrivateCallRequest struct and we validate that? If the oracle returns a call stack item that is largely unconstrained somebody could modify the above code to trust the oracle response when only part of it is constrained (the items that end up in PrivateCallRequest). Previously it was constrained because the whole of it was hashed

Reply:
But we are also getting return values and such from it. Think we might look at it in a separate go?